### PR TITLE
[GEOT-6703] Dwithin old syntax no longer supported

### PR DESCRIPTION
**Pull request by alemarte**
<Include a few sentences describing the overall goals for this Pull Request>

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geotools/geotools/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation) (I'm working in GeoSolutions)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.
- [ ] The changes are not breaking the build in downstream projects using SNAPSHOT dependencies, GeoWebCache and GeoServer (there is an automatic PR check verifying this, check this when it turns green).

The following are required only for core and extension modules (they are welcomed, but not required, for unsupported modules):
- [x] There is an issue in [Jira](https://osgeo-org.atlassian.net/projects/GEOT) describing the bug/task/new feature (a notable exemptions is, changes not visible to end users). The ticket is for the GeoTools project, if the issue was found elsewhere it's a good practice to link to the origin ticket/issue.
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message(s) must be in the form "[GEOT-XYZW] Title of the Jira ticket"
- [x] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Documentation has been updated accordingly.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

**Comment by alemarte** [comment]
Caused by https://github.com/geotools/geotools/pull/2734

The new DistanceUnits Binder is not hitted when performing a GetFeature query as the following one (introduced in a junit test).

<?xml version="1.0" encoding="UTF-8"?>
<wfs:GetFeature service="WFS" version="2.0.0"
	xmlns:wfs='http://www.opengis.net/wfs/2.0'
	xmlns:fes='http://www.opengis.net/fes/2.0'
	xmlns:gml='http://www.opengis.net/gml/3.2'
	xmlns:tiger='http://www.census.gov'>
	<wfs:Query typeNames="tiger:tiger_roads">
		<fes:Filter>
			<fes:DWithin>
				<fes:ValueReference>the_geom</fes:ValueReference>
				<gml:Point srsName="urn:ogc:def:crs:EPSG:3067">
					<gml:pos>219158.795 6800828.901</gml:pos>
				</gml:Point>
				<fes:Distance units="m">100</fes:Distance>
			</fes:DWithin>
		</fes:Filter>
	</wfs:Query>
</wfs:GetFeature>

The previous implementation has been reintroduced as a fallback to avoid NullPointerException.

@aaime Can you please check and review, and maybe suggest a better approach?

**Comment by aaime** [comment]
Let's check the schemas at:

```xml
<xsd:complexType name="DistanceBufferType">
<xsd:complexContent>
<xsd:extension base="fes:SpatialOpsType">
<xsd:sequence>
<xsd:choice maxOccurs="2">
<xsd:element ref="fes:expression"/>
<xsd:any namespace="##other"/>
</xsd:choice>
<xsd:element name="Distance" type="fes:MeasureType"/>
</xsd:sequence>
</xsd:extension>
</xsd:complexContent>
</xsd:complexType>
```

and fes:MeasureType is:

```
<xsd:complexType name="MeasureType">
<xsd:simpleContent>
<xsd:extension base="xsd:double">
<xsd:attribute name="uom" type="fes:UomIdentifier" use="required"/>
</xsd:extension>
</xsd:simpleContent>
</xsd:complexType>
```

So yes, the request is not schema valid, should have been:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<wfs:GetFeature service="WFS" version="2.0.0"
	xmlns:wfs='http://www.opengis.net/wfs/2.0'
	xmlns:fes='http://www.opengis.net/fes/2.0'
	xmlns:gml='http://www.opengis.net/gml/3.2'
	xmlns:tiger='http://www.census.gov'>
	<wfs:Query typeNames="tiger:tiger_roads">
		<fes:Filter>
			<fes:DWithin>
				<fes:ValueReference>the_geom</fes:ValueReference>
				<gml:Point srsName="urn:ogc:def:crs:EPSG:3067">
					<gml:pos>219158.795 6800828.901</gml:pos>
				</gml:Point>
				<fes:Distance uom="m">100</fes:Distance>
			</fes:DWithin>
		</fes:Filter>
	</wfs:Query>
</wfs:GetFeature>
```

The bug seems to be in the client making the request here?

**Comment by alemarte** [comment]
We restore the old tolerance for request for wfs 1.0 and 1.1 without any distance units specified.

The FES implementation will accept requests with the uom attribute of the Distance elements as per specification.

File: modules/extension/xsd/xsd-filter/src/main/java/org/geotools/filter/v1_0/DistanceUnits.java [hunk 1]
```diff
@@ -35,9 +35,11 @@ private DistanceUnits(double distance, String units) {
         if (distance < 0) {
             throw new IllegalArgumentException("distance must be >= 0");
         }
-        if (units == null) {
-            throw new IllegalArgumentException("units must not be null");
-        }
+
+        // We still support old client queries without any units specified.
```
Comment (alemarte) [inline modules/extension/xsd/xsd-filter/src/main/java/org/geotools/filter/v1_0/DistanceUnits.java:39]: The DistanceUnits constructor has been relaxed allowing null units.

File: modules/extension/xsd/xsd-fes/src/main/java/org/geotools/filter/v2_0/bindings/MeasureTypeBinding.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+ *  &lt;xsd:simpleType name="UomSymbol"&gt;
+ *      &lt;xsd:restriction base="xsd:string"&gt;
+ *          &lt;xsd:pattern value="[^: \n\r\t]+"/&gt;
+ *      &lt;/xsd:restriction&gt;
+ *  &lt;/xsd:simpleType&gt;
+ *
+ *         </code>
+ *  </pre>
+ *
+ * @generated
```
Comment (aaime) [inline modules/extension/xsd/xsd-fes/src/main/java/org/geotools/filter/v2_0/bindings/MeasureTypeBinding.java:58]: Has it actually been generated?

File: modules/extension/xsd/xsd-filter/src/main/java/org/geotools/filter/v1_0/DistanceUnits.java (hunk 1 above)
Comment (aaime) [reply modules/extension/xsd/xsd-filter/src/main/java/org/geotools/filter/v1_0/DistanceUnits.java:39]: Please remove the commented out code, the comment could use a small update too. E.g.:

```suggestion
        // We still support old client queries without any units specified, thus allowing units to be null.
```

**Review by aaime — COMMENTED** [review]
Looks good, requested minor updates.

File: modules/extension/xsd/xsd-fes/src/main/java/org/geotools/filter/v2_0/bindings/MeasureTypeBinding.java (hunk 2 above)
Comment (alemarte) [reply modules/extension/xsd/xsd-fes/src/main/java/org/geotools/filter/v2_0/bindings/MeasureTypeBinding.java:58]: Actually, I copy paste from the definition. Is there a tool to automate the process?

Comment (aaime) [reply modules/extension/xsd/xsd-fes/src/main/java/org/geotools/filter/v2_0/bindings/MeasureTypeBinding.java:58]: There is but only useful to get started. When you copy/paste, mind what you're copying. If not generated, the annotation should be removed.

**Comment by alemarte** [comment]
All the suggested minor updates have been committed.

**Comment by aaime** [comment]
All green, excellent. Merging, please do backports PRs as necessary.

**Pull request merged**